modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/BrowsersService.java

```diff
@@ -158,19 +188,55 @@ public void sendToAgent(AgentKey key, WebSocketRequest evt) {
                 case NODE_REST:
                 case NODE_VISOR:
                     JsonObject payload = fromJson(evt.getPayload());
+                    JsonObject params = payload.getJsonObject("params");
 
                     String clusterId = payload.getString("clusterId");
 
+                    if (evt.getEventType().equals(NODE_VISOR))
+                        params = buildRestExeParams(clusterId, params);
+
                     if (F.isEmpty(clusterId))
                         throw new IllegalStateException(messages.getMessage("err.missing-cluster-id-param"));
 
-                    if (evt.getEventType().equals(NODE_VISOR))
-                        evt.setPayload(toJson(fillVisorGatewayTaskParams(payload)));
+                    TopologySnapshot top = clustersRepo.get(clusterId);
+
+                    if (top == null)
+                        throw new IllegalStateException(messages.getMessageWithArgs("err.cluster-not-found-by-id", clusterId));
+
+                    // TODO GG-24427 Append sessionId if it exists in session store.
+                    if (top.isSecured()) {
+                        SessionAttribute sesAttr = new SessionAttribute(ses, SESSION_TOKEN_ATTR_PREFIX + clusterId);
+
+                        String sesTok = sesSrvc.get(sesAttr);
+
+                        if (F.isEmpty(sesTok) && Stream.of("user", "password").noneMatch(params::containsKey)) {
+                            sendMessageQuiet(ses, evt.response(RestResult.fail(STATUS_FAILED, "Failed to handle request - unknown session token (maybe expired session)")));
```

Comment:
Message "Failed to handle request..." should be in resources.

Reply:
No we just emulate same error in cluster

Reply:
Use value from static variable